smoke/src/main/java/com/adobe/cq/cloud/testing/it/smoke/ReplicationIT.java

```diff
@@ -15,178 +15,17 @@
-
-    @Rule
-    public CQRule cqBaseRule = new CQRule(cqBaseClassRule.authorRule, cqBaseClassRule.publishRule);
-
-    @Rule
-    public Page root = new Page(cqBaseClassRule.authorRule);
-
-    static CQClient adminAuthor;
-    static CQClient adminPublish;
-    static CQClient anonymousPublish;
-
-    private static ReplicationClient rClient;
-
-    @BeforeClass
-    public static void beforeClass() throws ClientException {
-        adminAuthor = cqBaseClassRule.authorRule.getAdminClient(CQClient.class);
-        adminPublish = cqBaseClassRule.publishRule.getAdminClient(CQClient.class);
-        anonymousPublish = cqBaseClassRule.publishRule.getClient(CQClient.class, null, null);
-        rClient = adminAuthor.adaptTo(ReplicationClient.class);
-    }
-
-    @Before
-    public void before() throws TimeoutException, InterruptedException {
-        new Polling(this::checkContentDistributionAgentExists)
-        	.poll(TIMEOUT, 500);
-    }
-
-	/**
-     * Activates a page as admin, then deactivates it. Verifies that the page gets removed from publish.
-     *
-     * @throws Exception if an error occurred
-     */
-    @Test
-    public void testActivateAndDeactivate() throws Exception {
-
-        // This is a workaround for correctly detecting assumption violations.
-        // Any AssumptionViolatedException throw by the the Callable will be
-        // swallowed by the Poller and wrapped in a TimeoutException. As a
-        // consequence, the test will be marked as failed instead of skipped.
-
-        try {
-            activateAndDeactivate();
-        } catch (AssumptionViolatedException e) {
-            throw e;
-        } catch (Exception e) {
-            new Polling(this::activateAndDeactivate).poll(TIMEOUT, 500);
-        }
-    }
-    
-    /**
-     * Activates a page as admin, than deletes it. Verifies that deleted page gets removed from publish.
-     *
-     * @throws Exception if an error occurred
-     */
-    @Test
-    public void testActivateAndDelete() throws Exception {
-
-        // This is a workaround for correctly detecting assumption violations.
-        // Any AssumptionViolatedException throw by the the Callable will be
-        // swallowed by the Poller and wrapped in a TimeoutException. As a
-        // consequence, the test will be marked as failed instead of skipped.
-
-        try {
-            activateAndDelete();
-        } catch (AssumptionViolatedException e) {
-            throw e;
-        } catch (Exception e) {
-            new Polling(this::activateAndDelete).poll(TIMEOUT, 500);
-        }
-    }
-    
-    private boolean activateAndDelete() throws Exception {
-        rClient.activate(root.getPath());
-        checkPage(SC_OK);
-
-        adminAuthor.deletePage(new String[]{root.getPath()}, true, false);
-        checkPage(SC_NOT_FOUND);
-        return true;
-    }
+/**
+ * Tests to check replication
+ */
+public class ReplicationIT extends PublishEndToEndIT {
```

Comment:
I don't think we should do this. No base classes for tests as things get confusing really quickly. Whenever we have shared code, we put this in `Rule`s or utility classes.

Reply:
Hmm...I think we will still have lot of shared code e.g the test methods and the internal methods they delegate to. Is that  acceptable? Here it ensures there is no duplication

Reply:
Yes, it's ok to have shared code, but not through inheritance :)

Reply:
Sure. what I mean is that we will have duplicated code e.g. the methods below can't logically be placed anywhere common other than a test but I guess that's ok.
- testActivateAndDelete
- activateAndDelete

Reply:
You can put them in a `Rule` class or a utility class, you don't need to duplicate the code